tests/test_inotify_c.py

```diff
@@ -187,3 +187,25 @@ def test_watch_file():
         os.remove(path)
         event, _ = event_queue.get(timeout=5)
         assert repr(event)
+
+
+def test_replace_moved_path():
+    from queue import Empty
+    from .shell import mkfile, mkdir, mv, cd, pwd
+
+    curdir = pwd()
+    tmpdir = p("")
+    cd(tmpdir)
+    mkdir("foo/foo/foo/foo", True)
+
+    with watching("foo"):
+        mv("foo/foo", "foo/bar")
```

Comment:
I kept getting Error at this line, so not able to verify that this test works as expected. Not sure whether this is because I am using Linux subsystem on Windows. Could you try whether it works for you?

Reply:
I rewrote the test like:

```python
def test_replace_moved_path():
    from queue import Empty
    from .shell import mkdir, mv

    path = p("foo", "foo", "foo", "foo")
    mkdir(path, parents=True)
    with watching(p()):
        mv(p("foo", "foo"), p("foo", "bar"))
        event, _ = event_queue.get(timeout=5)
        while True:
            try:
                event_queue.get(timeout=5)
                print(event)
            except Empty:
                break
        assert p("foo", "foo", "foo").encode() in emitter._inotify._inotify._wd_for_path
        assert p("foo", "bar", "foo", "bar").encode() not in emitter._inotify._inotify._wd_for_path
```

But that does not trigger the changed code at all :thinking: